CMakeLists.txt

```diff
@@ -19,6 +19,22 @@ TRY YOUR LUCK
 #TODO Add pcre support
 #TODO exclude tests which are not supported
 
+#NOTE On Windows when compiling with MSVC or Clang you need add invalidcontinue.obj to the link options
+#This is VERY VERY IMPORTANT!!!
```

Comment:
This should probably be squashed into the first commit.

Unless you want to structure the commits such that an initial `CMakeLists.txt` supports only Linux (possibly splitting out the `BUILD_TESTING` part into its own commit, and maybe even also the `install` part), then the macOS-specific parts are added, then the Windows ones? I would be a fan of this idea, as it would make it much easier to review this work on the Git mailing list (where I want to send this eventually).